xdsl/utils/mlir_lexer.py

```diff
@@ -170,41 +170,14 @@ class MLIRTokenKind(Enum):
     FILE_METADATA_BEGIN = "{-#"
     FILE_METADATA_END = "#-}"
 
-    @staticmethod
-    def get_punctuation_spelling_to_kind_dict() -> dict[str, MLIRTokenKind]:
-        return {
```

Comment:
creates a new dictionary each time we check whether a token is punctuation then throws it away